File: src/main/java/cn/labzen/logger/library/tomcat/TomcatLog.java
package cn.labzen.logger.library.tomcat;

import cn.labzen.logger.Loggers;
import cn.labzen.logger.kernel.LabzenLogger;
import org.apache.juli.logging.Log;

public class TomcatLog implements Log {

  private static final String SERVLET_NAME = "TOMCAT";
  private final LabzenLogger logger;

  public TomcatLog(String fqcn) {
    this.logger = Loggers.getLogger(fqcn);
  }

  public TomcatLog() {
    this("<UNKNOWN>");
  }

  private String safeToString(Object o) {
    return o == null ? "null" : o.toString();
  }

  @Override
  public boolean isDebugEnabled() {
    return logger.isDebugEnabled();
  }

  @Override
  public boolean isErrorEnabled() {
    return logger.isErrorEnabled();
  }

  @Override
  public boolean isFatalEnabled() {
    return logger.isErrorEnabled();
  }

  @Override
  public boolean isInfoEnabled() {
    return logger.isInfoEnabled();
  }

  @Override
  public boolean isTraceEnabled() {
    return logger.isTraceEnabled();
  }

  @Override
  public boolean isWarnEnabled() {
    return logger.isWarnEnabled();
  }

  @Override
  public void trace(Object o) {
    logger.atTrace().scene(SERVLET_NAME).log(safeToString(o));
  }

  @Override
  public void trace(Object o, Throwable throwable) {
    logger.atTrace().scene(SERVLET_NAME).setCause(throwable).log(safeToString(o));
  }

  @Override
  public void debug(Object o) {
    logger.atDebug().scene(SERVLET_NAME).log(safeToString(o));
  }

  @Override
  public void debug(Object o, Throwable throwable) {
    logger.atDebug().scene(SERVLET_NAME).setCause(throwable).log(safeToString(o));
  }

  @Override
  public void info(Object o) {
    logger.atInfo().scene(SERVLET_NAME).log(safeToString(o));
  }

  @Override
  public void info(Object o, Throwable throwable) {
    logger.atInfo().scene(SERVLET_NAME).setCause(throwable).log(safeToString(o));
  }

  @Override
  public void warn(Object o) {
    logger.atWarn().scene(SERVLET_NAME).log(safeToString(o));
  }

  @Override
  public void warn(Object o, Throwable throwable) {
    logger.atWarn().scene(SERVLET_NAME).setCause(throwable).log(safeToString(o));
  }

  @Override
  public void error(Object o) {
    logger.atError().scene(SERVLET_NAME).log(safeToString(o));
  }

  @Override
  public void error(Object o, Throwable throwable) {
    logger.atError().scene(SERVLET_NAME).setCause(throwable).log(safeToString(o));
  }

  @Override
  public void fatal(Object o) {
    logger.atError().scene(SERVLET_NAME).log(safeToString(o));
  }

  @Override
  public void fatal(Object o, Throwable throwable) {
    logger.atError().scene(SERVLET_NAME).setCause(throwable).log(safeToString(o));
  }
}

File: src/main/kotlin/cn/labzen/logger/library/tomcat/TomcatLog.kt
package cn.labzen.logger.library.tomcat

import cn.labzen.logger.Loggers
import org.apache.juli.logging.Log

class TomcatLog(fqcn: String) : Log {

  constructor() : this("_UNKNOWN_")

  private val logger = Loggers.getLogger(fqcn)

  companion object {
    private const val SERVLET_NAME = "TOMCAT"
  }

  override fun isDebugEnabled(): Boolean =
    logger.isDebugEnabled

  override fun isErrorEnabled(): Boolean =
    logger.isErrorEnabled

  override fun isFatalEnabled(): Boolean =
    logger.isErrorEnabled

  override fun isInfoEnabled(): Boolean =
    logger.isInfoEnabled

  override fun isTraceEnabled(): Boolean =
    logger.isTraceEnabled

  override fun isWarnEnabled(): Boolean =
    logger.isWarnEnabled

  override fun trace(message: Any?) {
    logger.trace().scene(SERVLET_NAME).log("$message")
  }

  override fun trace(message: Any?, t: Throwable?) {
    logger.trace().scene(SERVLET_NAME).setCause(t).log("$message")
  }

  override fun debug(message: Any?) {
    logger.debug().scene(SERVLET_NAME).log("$message")
  }

  override fun debug(message: Any?, t: Throwable?) {
    logger.debug().scene(SERVLET_NAME).setCause(t).log("$message")
  }

  override fun info(message: Any?) {
    logger.info().scene(SERVLET_NAME).log("$message")
  }

  override fun info(message: Any?, t: Throwable?) {
    logger.info().scene(SERVLET_NAME).setCause(t).log("$message")
  }

  override fun warn(message: Any?) {
    logger.warn().scene(SERVLET_NAME).log("$message")
  }

  override fun warn(message: Any?, t: Throwable?) {
    logger.warn().scene(SERVLET_NAME).setCause(t).log("$message")
  }

  override fun error(message: Any?) {
    logger.error().scene(SERVLET_NAME).log("$message")
  }

  override fun error(message: Any?, t: Throwable?) {
    logger.error().scene(SERVLET_NAME).setCause(t).log("$message")
  }

  override fun fatal(message: Any?) {
    logger.error().scene(SERVLET_NAME).log("$message")
  }

  override fun fatal(message: Any?, t: Throwable?) {
    logger.error().scene(SERVLET_NAME).setCause(t).log("$message")
  }
}
